first beta of MEI Basic

kepper opened the pull request:
this adds a draft for MEI basic, based on our discussions in Oxford. It's still not complete and needs significant consideration of:

- how much metadata should be included
- make sure controlevents have one method for placement only (but potentially a different for every type)

We could continue to develop in this branch, or merge it into develop.

kepper committed:
first draft of MEI basic, still incomplete

kepper committed:
schema derived from first version

lpugin commented on customizations/mei-basic.xml:
```diff
@@ -0,0 +1,405 @@
+                </constraintSpec>
+                
+                <moduleRef key="MEI"/>
+                <moduleRef key="MEI.cmn" except="beamSpan bend bracketSpan meterSig meterSigGrp mSpace oLayer ossia oStaff tupletSpan"/>
```
not sure about `meterSig` being excluded.

kepper commented:
The spreadsheet isn’t addressing everything, but we made a statement there about meterSig to be excluded. I think the idea was to encourage people to use scoreDef between measures instead. Basic should exclude meters changing in the middle of a measure (no clue how this would work anyway…)

lpugin commented:
> The spreadsheet isn’t addressing everything, but we made a statement there about meterSig to be excluded. I think the idea was to encourage people to use scoreDef between measures instead. Basic should exclude meters changing in the middle of a measure (no clue how this would work anyway…)

Makes sense. Is `keySig` excluded as well? It should be

kepper commented:
If I remember correctly (in need of a power outlet to verify…) the argument to keep keySig was that keys may actually change in the middle of a measure. However, we should make sure that it’s not used inside scoreDef…

ahankinson commented:
I would like to see us address #582 first before merging this PR

kepper committed:
Merge pull request #587 from music-encoding/develop

Release 4.0.1

kepper committed:
first beta version of MEI Basic

please test thoroughly

kepper commented:
We now have a first beta of MEI Basic. It hasn't been tested to great detail yet, though, so please have a look. I'm particularly curious to hear from @rettinghaus and @lpugin if it has everything that's needed for rendering with Meiler and Verovio, respectively.

kepper committed:
some more restrictions to MEI basic

kepper committed:
MEI Basic RC1

This is the first Release Candidate for MEI Basic (formerly known as MEI Go).

kepper committed:
update meiversion to 4.0.1-rc1+basic

kepper commented:
Dear @music-encoding/core-contributors, dear all, I finally managed to get around with the draft of MEI Basic, as we've discussed in the past. Please have a look and see, I surely missed some aspects. It is pretty restrictive, but is build to allow integration of additional features in the future without causing many disruptions -> very often, we picked elements, but disallowed all advanced attributes on them for the time being. For the `@meiversion`-attribute, the allowed value is `4.0.1-rc1+basic`. I like the `+basic` part at the end (which is a possible part in semantic versioning), because it allows us to indicate MEI profiles in the schema. We may even have multiple profile combinations (something like cmn + mensural), and control them through a regex, instead of a value list as we have right now. However, that'll be a separate issue to discuss… 

kepper commented:
btw., could someone more familiar with Travis look into the build failures? They don't seem to originate from problems in the repo, but in the Travis setup. @ahankinson, are you able to help out?

kepper committed:
Merge remote-tracking branch 'origin/master' into basic

martha-thomae commented:
> btw., could someone more familiar with Travis look into the build failures? They don't seem to originate from problems in the repo, but in the Travis setup. @ahankinson, are you able to help out?

I think the Travis issue is the one I fixed on another pull request https://github.com/music-encoding/music-encoding/pull/634/commits/4fbb8315501a2532d0b77086956c9a7b7865b6f3. I don't know Travis, this was with help of someone from my Lab that does. But it built ok afterward.

lpugin reviewed (commented):
`@stem.dir` missing on `<chord>`
`@fontname` missing on `<rend>`
`@accid` is allowed on `<note>` but not `@accid.ges`
`@vgrp` missing on `<dynam>` (I would still consider basic, but maybe not everybody would agree)
`@oloc` and `@ploc` on `<rest>` (ditto)

kepper committed:
address issues by @lpugin

This resolves the stem.dir issue, re-allows fontname, and takes care of accid (which is supposed to be an element anyway). I did not allow vgrp and oloc, though. Those can be added when there is more substantial demand…

lpugin commented:
For evaluating it with MNX, we should use the test files provided in their converter https://github.com/w3c/mnxconverter/tree/master/tests
(We could easily have an importer in Verovio for them. To test with MEI basic we should add an option to set a specific schema in the MEI output, but this is straighforward)
